internal/spoke_k8s_client/spoke_k8s_client_manager.go

```diff
@@ -0,0 +1,44 @@
+package spoke_k8s_client
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// SpokeK8sClientManager maintains a set of spoke clients
+
+//go:generate mockgen --build_flags=--mod=mod -package=spoke_k8s_client -destination=mock_spoke_k8s_client_manager.go . SpokeK8sClientManager
+type SpokeK8sClientManager interface {
+	AddFromSecret(secret *corev1.Secret) (SpokeK8sClient, error)
+}
+
+type spokeK8sClientManager struct {
+	clientFactory SpokeK8sClientFactory
+	clientMap     map[string]SpokeK8sClient
+}
+
+func NewSpokeK8sClientManager(log logrus.FieldLogger) SpokeK8sClientManager {
```

Comment:
A [general golang naming convention](https://go.dev/doc/effective_go#package-names) is to avoid stuttering in names. Here this will get used as `spoke_k8s_client.NewSpokeK8sClientManager()`. That would be better as `spoke_k8s_client.NewManager()`

FWIW there's also a convention to avoid snake-cases in package names and otherwise, but that seems to have appeared in a prior recent PR.

Reply:
Sure, makes sense. Moved to controllers package, so it'd be controllers.NewSpokeClientCache